src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HPack/Huffman.cs

```diff
@@ -308,119 +284,152 @@ public static (uint encoded, int bitLength) Encode(int data)
         /// <returns>The number of decoded symbols.</returns>
         public static int Decode(byte[] src, int offset, int count, byte[] dst)
         {
-            int i = offset;
-            int j = 0;
-            int lastDecodedBits = 0;
-            while (i < count)
+            // input validation
+            if (src == null || count == 0)
+                return 0;
+
+            if (dst == null)
+                throw new ArgumentNullException(nameof(dst));
+
+            if ((uint)offset >= (uint)src.Length)
```

Comment:
You can get rid of one of the throws by using `||`, reducing code size.

Reply:
You mean on the `ArgumentOutOfRangeException`? Wouldn't we need to keep two to preserve the difference between `nameof(offset)` and `nameof(count)`?

```C#
if ((uint)offset >= (uint)src.Length)
	throw new ArgumentOutOfRangeException(nameof(offset));
if ((uint)count > (uint)(src.Length - offset))
	throw new ArgumentOutOfRangeException(nameof(count));
```

Reply:
I need to clean my glasses 😸

Reply:
No problem. I thought maybe I was missing something.